civicrm_resources/civicrm_resources.module

```diff
@@ -6,205 +6,43 @@
- *     to 0.
- * @param $depth
- *   Current depth of recursion. This parameter is only used internally and
- *   should not be passed in.
- *
- * @return
- *   An associative array (keyed on the chosen key) of objects with 'uri',
- *   'filename', and 'name' members corresponding to the matching files.
+ * @param string $extension_path
+ *   Machine path of civicrm extension.
+ * @param array $resources_uris
+ *   An array of resources URIs relative to extension paths to be included
  */
-function civicrm_resources_file_scan_directory($dir, $mask, $options = [], $depth = 0) {
-  // Merge in defaults.
-  $options += [
-    'nomask' => '/(\.\.?|CVS)$/',
-    'callback' => 0,
-    'recurse' => TRUE,
-    'key' => 'uri',
-    'min_depth' => 0,
-  ];
-
-  $options['key'] = in_array($options['key'], [
-    'uri',
-    'filename',
-    'name',
-  ]) ? $options['key'] : 'uri';
-  $files = [];
-  if (is_dir($dir) && $handle = opendir($dir)) {
-    while (FALSE !== ($filename = readdir($handle))) {
-      if (!preg_match($options['nomask'], $filename) && $filename[0] != '.') {
-        $uri = "$dir/$filename";
-        $uri = file_stream_wrapper_uri_normalize($uri);
-        if (is_dir($uri) && $options['recurse']) {
-          $files = array_merge(civicrm_resources_file_scan_directory($uri, $mask, $options, $depth + 1), $files);
-        }
-        elseif ($depth >= $options['min_depth'] && preg_match($mask, $filename)) {
-          $file['uri'] = $uri;
-          $file['filename'] = $filename;
-          $file['name'] = pathinfo($filename, PATHINFO_FILENAME);
-          $key = $options['key'];
-          $files[$file[$key]] = $file;
-          if ($options['callback']) {
-            $options['callback']($uri);
-          }
-        }
-      }
-    }
-    closedir($handle);
+function civicrm_resources_add_resources($extension_path, $resources_uris = []) {
+  foreach ($resources_uris as $resources_uri) {
+    _civicrm_resources_add_resource($extension_path, $resources_uri);
   }
-  return $files;
 }
 
 /**
- * API function to be used across drupal site to load resources.
+ * Loads resources from the given extension
  *
- * @param string $extention_full_name
- *   Machine name of civicrm extention.
+ * @param string $extension_full_name
+ *   Machine name of civicrm extension.
  * @param array $files
- *   An array of file list to be included
- *
- * @return NULL
+ *   An array of resources URIs relative to extension paths to be included
  */
-function civicrm_resources_load($extention_full_name, $files = []) {
-  if (cache_get($extention_full_name, 'cache_civicrm_resources')) {
-    $ext_path_cache = cache_get($extention_full_name, 'cache_civicrm_resources');
+function civicrm_resources_load($extension_full_name, $resources_uris = []) {
+  if (cache_get($extension_full_name, 'cache_civicrm_resources')) {
+    $ext_path_cache = cache_get($extension_full_name, 'cache_civicrm_resources');
     $ext_path = $ext_path_cache->data;
   }
```

Comment:
I was checking how civi resolves the path and it looks like it already does some caching internally, so I think that having this additional cache here it's actually hurting the performance. Could you please remove it?

Reply:
I checked:

|  | Current PR code | What is suggested |
| ------------- | ------------- | ------------- |
| After cache flush | 0.0013080 | 0.0001314 |
| Second load | 0.0006390 | 0.0001256 |